src/app/clusters/barrier-control-server/barrier-control-server.cpp

```diff
@@ -67,7 +67,7 @@ static State state;
 // -----------------------------------------------------------------------------
 // Framework initialization
 
-void emberAfPluginBarrierControlServerInitCallback(void) {}
```

Comment:
Why don't we have no extern "C" declaration for these functions in scope at the definition? Are we missing an #include here?

It's important to have the declaration in scope before making the definition, otherwise the compiler cannot check that the declaration and definition have compatible signatures.

Reply:
We could pull in the header that declares this, yes.  I don't know why the original silabs code doesn't, but it's not a big deal to do it.

Reply:
That said, some of the other functions like this (e.g. `emberAfClusterSpecificCommandParse`) aren't in any headers right now...  I guess we can add them to headers.

Reply:
And actually, this specific one has the same problem; it's in a header but only as part of a weird define in generated headers.  I filed https://github.com/project-chip/connectedhomeip/issues/3619 to sort this out, but would really rather not block this PR on it if we can avoid it; it's already blocking multiple other things I am trying to do...

Reply:
@mspang I went through the places where I added `extern "C"` here, and they're all in that bucket.  We really need to start working on refactoring this code's header structure (and renaming things, while we're at it)....